xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/test/java/org/xwiki/refactoring/script/RefactoringScriptServiceTest.java

```diff
@@ -363,4 +367,30 @@ public void changeDocumentAuthor() throws Exception
         verify(request).setReplaceDocumentAuthor(true);
         verify(request).setReplaceDocumentContentAuthor(true);
     }
+
+    @Test
+    void canSkipRecycleBinAllTrue()
+    {
+        when(this.documentAccessBridge.isAdvancedUser()).thenReturn(true);
+        when(this.configuration.canSkipRecyclebin()).thenReturn(true);
+        final boolean actual = this.refactoringScriptService.canSkipRecycleBin();
+        assertTrue(actual);
+    }
+
+    @Test
+    void canSkipRecycleBinConfigurationIsFalse()
+    {
+        when(this.configuration.canSkipRecyclebin()).thenReturn(false);
+        final boolean actual = this.refactoringScriptService.canSkipRecycleBin();
+        assertFalse(actual);
+    }
+
+    @Test
+    void canSkipRecycleBinAdvancedIsFalse()
```

Comment:
The naming could be improved IMO. For example `canSkipRecycleBinWhenSimpleUser`.

Same for the others.

Reply:
Test method names updated (see https://github.com/xwiki/xwiki-platform/pull/1369/files#diff-2abde92d350206b28dbfc6499abfabc8R372)